tests/agents/test_orchestrator.py

```
@@ -0,0 +1,366 @@
+        """Should gracefully handle coordination failures."""
+        if hasattr(orchestrator, 'coordinate'):
+            failing_agent = MagicMock()
+            failing_agent.id = "failing-agent"
+            failing_agent.execute = AsyncMock(side_effect=Exception("Agent failed"))
+
+            workflow = {
+                "steps": [{"agent_id": failing_agent.id, "action": "fail"}]
+            }
+
+            with patch.object(orchestrator, 'agents', {failing_agent.id: failing_agent}):
+                with pytest.raises(Exception) or result:
```

Flake8 (7.3.0)

[error] undefined name 'result'

(F821)